Userland/Libraries/LibSQL/AST/Lexer.cpp

```diff
@@ -93,17 +99,17 @@ Token Lexer::next()
             if (auto it = s_one_char_tokens.find(m_current_char); it != s_one_char_tokens.end()) {
                 found_one_char_token = true;
                 token_type = it->value;
-                consume();
+                consume(&current_token);
             }
         }
 
         if (!found_two_char_token && !found_one_char_token) {
             token_type = TokenType::Invalid;
-            consume();
+            consume(&current_token);
         }
     }
 
-    Token token(token_type, m_source.substring_view(value_start - 1, m_position - value_start), value_start_line_number, value_start_column_number);
+    Token token(token_type, current_token.string_view(), value_start_line_number, value_start_column_number);
```

Comment:
Probably better for this to be `current_token.build()`, which creates a `String` from the `StringBuilder`, rather than a view to its underlying buffer. Then the constructor for `Token` can take a `String` rather than a `StringView`.

Reply:
Yeah the whole StringBuilder/String/StringView/FlyString world is still a bit of a mystery to me sometimes. Will fix.